### PR TITLE
Fix image model

### LINKED ISSUE #959: Bug when running picca v6 with y1 blinding
I am running at NERSC in  `/global/cfs/cdirs/desi/users/jguy/lya/himalayas/`.

Successful run of
```
picca_delta_extraction.py picca-v6.0/deltas/delta_lya.ini
```
But crash at the next step
```
picca_cf.py --in-dir picca-v6.0/deltas/lya/Delta --out picca-v6.0/correlations/lya_x_lya/cf_lya_x_lya.fits --nside 16
ATTENTION: Analysis is blinded with strategy desi_y1
Om=0.315, Or=0.0, wl=-1.0, H0=100.0
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/global/common/software/desi/cori/desiconda/20211217-2.0.0/conda/lib/python3.9/site-packages/fitsio/hdu/table.py", line 1745, in _extract_colnum
    colnum = self._colnames_lower.index(colstr.lower())
ValueError: 'delta_blind' is not in list
...
```

### LINKED ISSUE #960: Missing option --from-image for picca_dmat.py
Missing option --from-image for picca_dmat.py prevents to compute the distortion matrix with the new delta format.

### LINKED ISSUE #961: picca_xcf.py failing with new deltas
picca_xcf.py --new-image failing with new delta format.

It's reading the deltas with the old format despite the use of option '--new-image '. Maybe I don't understand correctly how this should be run. I tried:
```
cd /global/cfs/cdirs/desi/users/jguy/lya/himalayas/

picca_xcf.py --in-dir picca-v6.0/deltas/lya/Delta --drq catalogs/QSO_catalog_dark_healpix_lastnight_lya.fits --in-dir picca-v6.0/deltas/lya/Delta --out picca-v6.0/correlations/lya_x_qso/cf_lya_x_qso.fits --nside 16 --from-image --mode desi_healpix
```

### PR DESCRIPTION
**Pull request by cramirezpe**
This fixes #960 and fixes #961 and fixes #959. Now it is not needed to specify format.

I've also added log_lambda_index into mask fields so it gets masked when masks are applied.

### CODE REVIEW

**Comment by cramirezpe** [comment]
I had problems fixing forest.log_lambda_index. When we first introduced this parameter, we substituted some code for Pk1d in order to use this new attribute instead of returning bins between methods. We made all the modifications without affecting tests for Pk1d, so in principle we did not break anything.

Now that we are updating forest.log_lambda_bins so it uses rebined bins, I have found that Pk1d tests fail. I needed to modify Pk1d code to use the returned original bins in order to make the tests work again.

I think someone from the Pk1d team should take a look at this to ensure everything is fine (although this can be merged as long as tests work).

**Review by iprafols — APPROVED** [review]

**Pull request merged**
